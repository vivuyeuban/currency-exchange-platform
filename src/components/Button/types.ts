import { ButtonProps } from '@material-ui/core';

export type TButtonProps = ButtonProps;
