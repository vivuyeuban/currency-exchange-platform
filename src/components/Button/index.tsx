import React from 'react';
import Button from '@material-ui/core/Button';
import type { TButtonProps } from './types';

const MyButton = React.memo(({
  children,
  ...props
}: TButtonProps) => (
  <Button
    color="primary"
    variant="contained"
    {...props}
  >
    {children}
  </Button>
));

export default MyButton;
