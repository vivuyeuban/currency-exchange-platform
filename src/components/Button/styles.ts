import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import pink from '@material-ui/core/colors/pink';
import orange from '@material-ui/core/colors/orange';
import { grey } from '@material-ui/core/colors';

export const useStyles = makeStyles((theme: Theme) => createStyles({
  root: {
    background: `linear-gradient(45deg, ${pink[300]} 30%, ${orange[300]} 90%)`,
    borderRadius: 10,
    height: 40,
    padding: '0 30px',
  },
  disabled: {
    background: `linear-gradient(45deg, ${grey[700]} 30%, ${grey[500]} 90%)`,
  },
  label: {
    color: theme.palette.common.white,
  },
}));
