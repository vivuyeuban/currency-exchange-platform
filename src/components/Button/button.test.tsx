import React from 'react';
import { render } from '@testing-library/react';
import Button from './index';

test('renders button', () => {
  const { container } = render(<Button />);
  expect(container).toMatchSnapshot();
});

test('renders button with children', () => {
  const { container } = render(<Button>Custom Text</Button>);
  expect(container).toHaveTextContent('Custom Text');
});
