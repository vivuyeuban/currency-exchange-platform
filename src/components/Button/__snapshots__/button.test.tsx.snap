// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
